src/components/panel/panel.spec.js

```diff
@@ -117,6 +120,110 @@ describe('$mdPanel', function() {
     expect(parentEl.childElementCount).toEqual(0);
   });
 
+  describe('component logic: ', function() {
+    it('should allow templateUrl to specify content',
+        inject(function($templateCache) {
+          var htmlContent = 'Hello World';
+          var template = '<div id="panel">' + htmlContent + '</div>';
+          $templateCache.put('template.html', template);
+
+          var config = { templateUrl: 'template.html' };
+
+          panelRef = $mdPanel.open(config);
+          $rootScope.$apply();
+
+          expect('#panel').toContainHtml(htmlContent);
+        }));
+
+    it('should allow template to specify content', function() {
+      var htmlContent = 'Hello World';
+      var template = '<div id="panel">' + htmlContent + '</div>';
+      var config = { template: template };
+
+      panelRef = $mdPanel.open(config);
+      $rootScope.$apply();
+
+      expect('#panel').toContainHtml(htmlContent);
+    });
+
+    it('should allow a controller to be specified', function() {
+      var htmlContent = 'Hello World';
+      var template = '<div id="panel">{{ content }}</div>';
+
+      var config = {
+        template: template,
+        controller: function Ctrl($scope) {
+          $scope['content'] = htmlContent;
+        }
+      };
+
+      panelRef = $mdPanel.open(config);
+      $rootScope.$apply();
+
+      expect('#panel').toContainHtml(htmlContent);
+    });
+
+    it('should allow controllerAs syntax', function() {
+      var htmlContent = 'Hello World';
+      var template = '<div id="panel">{{ ctrl.content }}</div>';
+
+      var config = {
+        template: template,
+        controller: function Ctrl() {
+          this.content = htmlContent;
+        },
+        controllerAs: 'ctrl'
+      };
+
+      panelRef = $mdPanel.open(config);
+      $rootScope.$apply();
+
+      expect('#panel').toContainHtml(htmlContent);
+    });
+
+    it('should bind resolves to the controller', function() {
+      var htmlContent = 'Hello World';
+      var template = '<div id="panel">{{ ctrl.content }}</div>';
+
+      var config = {
+        template: template,
+        controller: function Ctrl() {
+          this.content; // Populated via bindToController.
+        },
+        controllerAs: 'ctrl',
+        resolve: {
+          content: function($q) {
+            return $q.when(htmlContent);
+          }
+        }
+      };
+
+      panelRef = $mdPanel.open(config);
+      $rootScope.$apply();
+
+      expect('#panel').toContainHtml(htmlContent);
+    });
+
+    it('should bind locals to the controller', function() {
+      var htmlContent = 'Hello World';
+      var template = '<div id="panel">{{ ctrl.content }}</div>';
+
+      var config = {
+        template: template,
+        controller: function Ctrl() {
```

Comment:
Should there also be a test where the controller injects (via constructor) something from locals?